juniper_hyper/src/lib.rs

```diff
@@ -215,45 +276,74 @@ where
+                    .map(move |request| {
                         let root_node = root_node.clone();
-                        let context = context.clone();
-                        future::poll_fn(move || {
-                            let res = futures::try_ready!(tokio_threadpool::blocking(
-                                || request.execute(&root_node, &context)
-                            ));
-                            let is_ok = res.is_ok();
-                            let body = serde_json::to_string_pretty(&res).unwrap();
-                            Ok(Async::Ready((is_ok, body)))
-                        })
-                    }))
-                    .map(|results| {
-                        let is_ok = results.iter().all(|&(is_ok, _)| is_ok);
-                        // concatenate json bodies as array
-                        // TODO: maybe use Body chunks instead?
-                        let bodies: Vec<_> = results.into_iter().map(|(_, body)| body).collect();
-                        let body = hyper::Body::from(format!("[{}]", bodies.join(",")));
+                        let res = request.execute(&root_node, &context);
+                        let is_ok = res.is_ok();
+                        let body = serde_json::to_string_pretty(&res).unwrap();
                         (is_ok, body)
-                    }),
-                )
+                    })
+                    .collect();
+
+                let is_ok = results.iter().all(|&(is_ok, _)| is_ok);
+                let bodies: Vec<_> = results.into_iter().map(|(_, body)| body).collect();
+                let body = hyper::Body::from(format!("[{}]", bodies.join(",")));
+                (is_ok, body)
+            }
+        }
+    }
+
+    #[cfg(feature = "async")]
+    async fn execute_async<'a, CtxT: 'a, QueryT, MutationT>(
+        self,
+        root_node: Arc<RootNode<'a, QueryT, MutationT, S>>,
+        context: Arc<CtxT>,
+    ) -> (bool, hyper::Body)
+    where
+        S: Send + Sync,
+        QueryT: GraphQLTypeAsync<S, Context = CtxT> + Send + Sync,
+        MutationT: GraphQLTypeAsync<S, Context = CtxT> + Send + Sync,
+        QueryT::TypeInfo: Send + Sync,
+        MutationT::TypeInfo: Send + Sync,
+        CtxT: Send + Sync,
+    {
+        match self {
+            GraphQLRequest::Single(request) => {
+                let res = request.execute_async(&root_node, &context).await;
+                let is_ok = res.is_ok();
+                let body = Body::from(serde_json::to_string_pretty(&res).unwrap());
+                (is_ok, body)
+            }
+            GraphQLRequest::Batch(requests) => {
+                let futures = requests
+                    .iter()
+                    .map(|request| request.execute_async(&root_node, &context))
+                    .collect::<Vec<_>>();
+                let results = futures::future::join_all(futures).await;
+
+                let is_ok = results.iter().all(|res| res.is_ok());
```

Comment:
Rather than checking for `all`, you can just look for one failure with `any`. Likely a  tiny bit faster in the case of an error.

Reply:
That was from the original code, but I fixed it anyways now.